pkg/ipam/node.go

```diff
@@ -546,8 +569,18 @@ func (n *Node) determineMaintenanceAction() (*maintenanceAction, error) {
 		return nil, err
 	}
 
+	surgeAllocate := 0
+	numPendingPods, err := getPendingPodCnt(n.name)
+	if err != nil {
+		scopedLog.WithError(err).Infof("Unable to compute pending pods, pod watcher might be initializing")
```

Comment:
I think we can probably make this a debug log line and remove the `pod watcher might be initializing` piece because the `err` will tell us the reason we are unable to compute pending pods. Instead, we could log that we were unable to compute pending pods, but the fallback is to allocate the number of needed IPs + the max above watermark.

```go
scopedLog.WithError(err).Infof("Unable to compute the number of needed IPs on a surge-basis IPs from pending pods, falling back to non-surge ...")
```

Something along those lines, feel free to word it.

Reply:
To add to what Chris wrote: Might also be worth adding some kind of rate limit to this error message. I think if a cluster has N nodes, this will potentially be printed N times if called before the watcher is initialized.

Reply:
That makes sense. @gandro is there a recommended way of limiting the number of times the same log line is printed ? Do we already do this elsewhere ? If not should i convert it to `debug` level ?

Reply:
Yes, kind of. We do have a logging rate limiter here:

https://github.com/cilium/cilium/blob/314ca7baeff4f568ffc0bad95124e40665b1f88c/pkg/logging/limiter.go#L23

Here is some example:
https://github.com/cilium/cilium/blob/314ca7baeff4f568ffc0bad95124e40665b1f88c/pkg/endpoint/policy.go#L45

Reply:
@gandro added log limiter to the `Node` struct and  updated the log line to use it.